doc/api/esm.md

```diff
@@ -431,6 +431,47 @@ thrown:
 }
 ```
 
+#### Self-referencing a package using its name
+
+Within a package, the values defined in the package’s
+`package.json` `"exports"` field can be referenced via the package’s name.
+For example, assuming the `package.json` is:
+
+```json
+// package.json
+{
+  "name": "a-package",
+  "exports": {
+    ".": "./main.mjs",
+    "./foo": "./foo.js"
+  }
+}
+```
+
+Then any module _in that package_ can reference an export in the package itself:
+
+```js
+// ./a-module.mjs
+import { something } from 'a-package'; // Imports "something" from ./main.mjs.
+```
+
+Self-referencing is available only if `package.json` has `exports`, and will
+allow importing only what that `exports` (in the `package.json`) allows.
+So the code below, given the package above, will generate a runtime error:
+
+```js
+// ./another-module.mjs
+import { another } from 'a-package/m.mjs'; // Imports "another" from ./m.mjs.
```

Comment:
Can the comment say that this errors? Usually we include that kind of thing in the comments even when the main text also says it, because many people skip there main text and only read the code examples.

Reply:
Done.